### ISSUE TITLE
Support for TV platforms

### ISSUE THREAD

**Issue by retardgerman** [fredrikburmester/streamyfin#137]
We should extend the support of the app to TV operating systems such as AndroidTV, AppleTV etc.

**Comment by rodrigogonegit** [comment]
Only if you want to become the de facto standard client 😀 

(At least in my household)

**Comment by tom-james-watson** [comment]
What's involved in making this happen? I ask because if there's a clear path for what needs implementing to make this a reality, it's more likely that other devs like myself might get involved. 🙏

**Comment by fredrikburmester** [comment]
Well, we're running Expo, which has some support for TVs. The hard part is adapting the UI to work with remotes, as well as removing all packages not supported on TV (I actually did install a native tab bar recently in our VLC branch which would make ATV easier). This is actually a much harder thing to do than you might think and will probably involve having a separate branch if Expo TV support doesn't get drastically better. 

If people want to help make this a reality I would encourage them to read up on Expo tv support, fork the repo, and try to get it running. I already made a tiny working example for ATV in a branch feat/tv-os, but it's old now.

I have never developer for Android TV so I can't really comment on that at all. 

There's no timeframe for this, as we have a lot of other bugs we have to fix right now, as well as getting the new VLC player out. 

@tom-james-watson

**Comment by eshirvana** [comment]
looking forward to this ....specially since there is no Tizen client app yet , that would be a great idea

**Comment by vttc08** [comment]
I also like Android TV version. For developing the ATV version, can I suggest implementing a feature for external video player support? Just like with jellyfin official client.

The built-in player will need to take account all ATV models, OS, codec support and compatibility. I'm certain on my Homatics Box R 4K plus android 12 it will have problems. Out of all the options for Android playback, Just Player with Jellyfin official app is the ONLY way I can use to play from jellyfin that doesn't crash, mark playback time, audio and subtitle support. But I really like the UI of streamyfin and integration with jellyseerr so I want to mention this.

**Comment by Zetto12** [comment]
Same here, 95% of my Jellyfin usage is on my Android TV. As soon as it shows up in the app store this will replace the jellyfin player app.

**Comment by damomato** [comment]
I agree with the above, I think there are a lot of people waiting on a good ATV client for Jellyfin and will get a lot of good use out of it.

I currently use Jellyfin Kodi add-on to watch through Kodi which works very well but would switch to a good Jellyfin client on Android TV if it had some of the newer features such as trickplay and skip intro etc. (though I did just stumbled upon the Jelly Skip add-on for Kodi I have yet to try). 

I also find Kodi a bit clunky and slow at times and a lot of work is needed to customise it to something worthwhile and I think I'm at the point in my life where I would prefer to use a proper app for Jellyfin on Android TV which seamlessly integrates with the server instead of relying on add-ons and customisation.

So I'm very excited to see work progress on this! Your work is very much appreciated.

**Comment by rottedmood** [comment]
huge plus 1 for AppleTV support. We really only use Jellyfin at home on our TV, and the other client we are using now is good, but tends to not show everything in the library (things like music videos, books, or other items don't tend to show up). I'm really stupid with coding, but i will def start to read up on expo and see what (if anything) i could help provide. The iOS is really, really well done, so would be cool to see equivalent on AppleTV

**Comment by devotoare** [comment]
Tizen support would be huge, given that Samsung TVs are so popular and don't support Chromecast.

**Comment by GrabbenD** [comment]
> Tizen support

Time- is expensive..
Chromecast (Android TV) isn't

From experience, Development for closed source TV OS with outdated and restrictive ecosystem is a work of pain

**Comment by damomato** [comment]
B

**Comment by devotoare** [comment]
> > Tizen support
> 
> Time is expensive.. Chromecast (Android TV) isn't
> 
> From experience, development in a closed source TV OS with outdated frameworks and a restrictive ecosystem is a world of pain

True. But pretending Samsung TV marketshare isn't huge and this wouldn't be worthwhile is weird.

**Comment by damomato** [comment]
Either way you can cheaply purchase an Android TV or Apple TV device to use with any TV, so I think for that accessibility that development should be the main focus for now.

**Comment by GrabbenD** [comment]
> True. But pretending Samsung TV marketshare isn't huge and this wouldn't be worthwhile is weird.

For a commercial service, yes that'd be weird.

PRs are welcome

**Comment by JamsRepos** [comment]
> Need this desperately! No real way to watch live tv through Jellyfin on AppleTV without using the half baked alpha in the Swiftfin app

_Slightly_ off-topic but in the meantime you can use Infuse, paid yes but a suitable alternative until Streamyfin steals the spotlight.

**Comment by timleeds15** [comment]
> > Need this desperately! No real way to watch live tv through Jellyfin on AppleTV without using the half baked alpha in the Swiftfin app
> 
> _Slightly_ off-topic but in the meantime you can use Infuse, paid yes but a suitable alternative until Streamyfin steals the spotlight.

I didn’t think Infuse supported Live TV no? Just local media?

**Comment by JamsRepos** [comment]
> > > Need this desperately! No real way to watch live tv through Jellyfin on AppleTV without using the half baked alpha in the Swiftfin app
> > 
> > 
> > _Slightly_ off-topic but in the meantime you can use Infuse, paid yes but a suitable alternative until Streamyfin steals the spotlight.
> 
> I didn’t think Infuse supported Live TV no? Just local media?

Apologies, didn't see "Live TV" in your response.

**Comment by DinleyH** [comment]
While i agree tizen tv support would be ideal it is a nightmare to work with and get approved, Jellyfin Samsung clients are web wrappers I think so if a browser web client is made it might be easier to look at it then. 

LG tv's are also quite widespread but they use webOS so not sure how workable that is 🤔 I've been looking into how expo could be made to work on webos and a desktop web version myself but I'm not familiar with expo so it's a bit of a learning curb 🤣 it would be really cool to see this app on TVs though as it's everything I've dreamed of since I got Jellyfin up and running. I'd love to contribute on a browser version.

**Comment by edouardkleinhans** [comment]
Did you plan to release app on amazon store (for fire tv) ?

**Comment by progecon** [comment]
+1 for amazon store!

**Comment by garret** [comment]
I am currently using Android Jellyfin TV client but would like to try Streamyfin. 

Today I tried to sideload the latest release .apk found in the release section on an Amazon Fire stick 4K Ultra HD but after I click Install (using Material Files as file manager) the installation process stops immediately. Is that expected? Is there another version .apk more tailored for Android TV that I should use?

**Comment by Simon-Eklundh** [comment]
Hi there @garret there's no TV apk at the moment, you'd need to build it yourself. That said: it's in a pre-alpha state and not really being actively worked on afaik

**Comment by theryecatcher** [comment]
AFAIK it's because the build is for arm64 while most if not all Android OS based TVs only support apps built for arm (32 bit).

**Comment by Simon-Eklundh** [comment]
That's possible, but it's also because we build TV with a special flag to remove Chromecast support

**Comment by sbkg0002** [comment]
> Hi there [@garret](https://github.com/garret) there's no TV apk at the moment, you'd need to build it yourself. That said: it's in a pre-alpha state and not really being actively worked on afaik

Is the `develop` branch the best branch to build to test this out?

**Comment by Simon-Eklundh** [comment]
I'd recommend develop generally, although occasionally it might break TV, if so, try master/main (I keep forgetting which)

**Comment by Simon-Eklundh** [comment]
@StatelessOddity i don't think that would be enough.

However, one of the lead Devs is swapping to ungoogled android soon™, so he'll be looking into doing a proper ungoogled Streamyfin build

**Comment by krishne35** [comment]
Is there any road map for android tv release?

**Comment by Simon-Eklundh** [comment]
Not really, basically it'll come when it comes :)

**Comment by progecon** [comment]
Just adding my name here, so you know that there are actually a few people looking forward to a TV version :)

**Comment by JamsRepos** [comment]
The contributors are fully aware of the interest in this. Repeated comments just flood the thread and add noise. Everyone here is a volunteer – the pace of work isn’t going to change based on how many people say they’re keen.

What would actually help right now is more contributors to share the load. These are people with jobs, lives, and limited time. Pressure or forced motivation doesn’t help – it creates stress and resentment. If expectations pile up, it’s more likely to push someone away than inspire them to keep going.

**Comment by DinleyH** [comment]
> > I think jellyfin for samsungs is done the same (or similar way) just a wrapper for the web client.
> 
> That's more or less how the other TV clients also work. So if you can build that, please do :)

Problem is streamyfin is only a mobile app right? no web-client so cant even get the tizen app to point to it as its not hosted like jellyfin and jellyseer are etc. If a web client is made then it would be easier.

**Comment by darkcloud784** [comment]
Any news on this? I would be stoked to see this on androidTV (My Nvidia Shield)

**Comment by MajorP93** [comment]
> Any news on this? I would be stoked to see this on androidTV (My Nvidia Shield)

Yes I am also really interested in Streamyfin on Android TV platform!

That would be an awesome addition to home streaming setups!

**Comment by Simon-Eklundh** [comment]
There is a sideloadable apk available nowadays

It's still in alpha but...

**Comment by garret** [comment]
> There is a sideloadable apk available nowadays
> 
> It's still in alpha but...

Where can it be found?

**Comment by Simon-Eklundh** [comment]
That's a gosh darn good question!

https://1drv.ms/u/c/ddb6042f501b7ed3/ETh-LAM0mIhIm7-PtVML6_sBsJVoHssyhCetZWUpe-h4Xw

That link might work

Otherwise you'll probably need to pop by the discord server (you can probably find it in the build artefacts for some commit on GitHub too)

**Comment by damomato** [comment]
Before I give that a shot, how well does it work? And is development active?

**Comment by Simon-Eklundh** [comment]
Development is fairly active (although currently we're working more on the downloads)

And functionality wise; it depends on your device I think, it worked well for some worse for others


My guess is if you have a smart TV it might be worse than a dedicated TV box

**Comment by damomato** [comment]
Thanks, I'll hold off for now as I have a good set up with Jelyfin for Kodi plugin, just would be nice to have something native with the skip intro and other Jellyfin specific features

**Comment by mfalkvidd** [comment]
Might be a good opportunity to mention https://github.com/streamyfin/streamyfin#-beta-testing

**Comment by sewersurfin** [comment]
Infuse needs a competitor!

**Comment by lancechant** [comment]
Just linking work for the tv stuff here:

Making it feel more user friendly: #1020 
Blocker: #1016 
#972

**Comment by TaakoMagnusen** [comment]
IN PROGRESS?! SO HYPE! I am frankly so sick and tired of the bugs and idiosyncracies with the official jellyfin android tv client 

Can't wait to try Streamyfin on my TV!

**Comment by damomato** [comment]
I agree, can't wait to move off Kodi

**Comment by sewersurfin** [comment]
Streamyfin on iOS is great so far. Seamless experience. Great integration of IPTV too. Adapting this to Apple TV would probably knock infuse—which has become stagnant developmentally—as the de facto go-to for Apple TV and provide a seamless, consistent solution for Jellyfin users. We need more customization and more integration of core Jellyfin features (eg, transcoding).

**Comment by lancechant** [comment]
There is a beta for Android TV already, just the Apple TV side is not building because of other packages breaking it, but hopefully soon. But just to set expectations, Tizen TV is not planned or achievable, because a lot of the packages aren't supporting web so it's not going to be anything close to the same experience or even gotten to.

**Comment by kauelima** [comment]
> There is a beta for Android TV already, just the Apple TV side is not building because of other packages breaking it, but hopefully soon. But just to set expectations, Tizen TV is not planned or achievable, because a lot of the packages aren't supporting web so it's not going to be anything close to the same experience or even gotten to.

Just for everyone to see, here it is the release with the beta if you want to test it and give feedback:

https://github.com/streamyfin/streamyfin/releases/tag/v0.30.2

**Comment by Darknior** [comment]
> https://github.com/streamyfin/streamyfin/releases/tag/v0.30.2

Thanks for the link, i have try it because i search for a real good TV player for me and my friends on Android TV ... i don't like watching films on phone or tablet, it's horrible :(
And before pay for the beta test, it is not a problem for me, i was try if it will works fine on my NVIDIA Shield Stick.
And it works ... but it is made for sure for phone, for touch screen :(
So complicated to use it with a remote :( many are impossible to use ...
I cant really choose audio track or subtitle track like it is possible on the web jellyfin interface ... or the Emby TV app :(

And the app also look for phone, it is not beutiful, watching medias in full screen like the web app ...
Don't use CSS too, watch the synopsis with only two lines ... so strange on a large TV screen ...
And wy not ... showing one day the MEDIA PLUS we have like CD, Blueray, spinning near the case ... like on XBMC.

You have made a fantastic work for now, for sure !!!!
I love your project, but i will must stand the v2 if i understand fine :(

**Comment by lancechant** [comment]
@Darknior there is a lot of work being done at the moment in a PR and will be a big improvement

**Comment by Darknior** [comment]
@lancechant so happy to read you, for sure i'm really hype to try it one day... and found my perfect TV app :)
I 'm really interested by this next improvement ... and them become beta tester because i only use TV :)
thanks for your message, i will stand for it

**Comment by viggy96** [comment]
> Yes, in the case of Streamyfin you are right, but the official Jellyfin TV client is just a wrapper.
> A Samsung Streamyfin version would be awesome, but I think that every Samsung owner will be happy with a working Jellyfin client for their set. Maybe not the right place to discuss this, but if it works, great!

The official Jellyfin TV app is a native Android app.

**Comment by milktoastrat** [comment]
I'd pay stupid money for a streamyfin apple TV app.

**Comment by nhabbash** [comment]
I'm also keeping an eye on this, would love to ditch Swiftfin or avoid having to buy Infuse to stream Jellyfin on Apple TV :)

**Comment by xorgx3** [comment]
And what about LG WebOs?

**Comment by Simon-Eklundh** [comment]
Webos is extremely unlikely I'd say

**Comment by damomato** [comment]
I just tried out Wholphin, and while it's great that others are working on android TV clients it's not good enough for me to move over from my current Kodi + Jellyfin add-on set up. 

One thing I noted though is that loading content was definitely much faster, which shows the promise this addition will have

**Comment by mxkyb** [comment]
Regarding WebOS, why not https://github.com/jellyfin/jellyfin-webos?

**Comment by xorgx3** [comment]
Because I'm looking for simple way to use Jellyserr with Jellyfin, and as far as I know only streamyfin supports that

**Comment by mxkyb** [comment]
@xorgx3 there is https://github.com/n00bcodr/Jellyfin-Enhanced, it might work for the webos client

**Comment by xorgx3** [comment]
Yeah, I know that very well, I even try contribute to that project but it realized impossible to do in webos. I consider switch to android tv box for better experiences

**Comment by damomato** [comment]
Just get a SHIELD and do away with webOS

**Comment by galdo320** [comment]
Hopping to see an Apple TV version soon.

**Comment by DinleyH** [comment]
> Yeah, I know that very well, I even try contribute to that project but it realized impossible to do in webos. I consider switch to android tv box for better experiences

Oh this interesting I’ve never seen Jellyfin enhanced !  I’m curious though, why is it unworkable with WebOS?

**Comment by rpcarvalheira** [comment]
Testing this and currently I have no clue how to open the menus using mi box s 3rd gen. 
Many things I'm unable to click and home is also unavailable unfortunately. 

Maybe I'm just dumb.

**Comment by Simon-Eklundh** [comment]
@rpcarvalheira I assume you're using the version in releases? It's not very well optimised for TV when it comes to control. There is another version hidden deep in the development channel on discord that has better controls for TV

**Comment by damomato** [comment]
Do you have a link to the APK for that? I couldn't get the version in the latest release to install

**Comment by Simon-Eklundh** [comment]
I believe this should work https://github.com/streamyfin/streamyfin/actions/runs/17856362066/artifacts/4055311355

**Comment by damomato** [comment]
Thanks. Unfortunately I'm getting a 404 error. I'm happy to wait until the next release if easier

**Comment by hillbicks** [comment]
> Thanks. Unfortunately I'm getting a 404 error. I'm happy to wait until the next release if easier

You have to be logged into github, otherwise you're getting the 404.

**Comment by damomato** [comment]
Ah thanks, on mobile and didn't realise. Downloaded now and will give it a shot.

Edit:
Ok installed it and it's working great, seems to load very fast. Well done to all who have worked on it. I can't see how to easily change users/profiles, but I'm guessing that's still to come.

My problem is that I'm comparing this to Kodi (with Jellyfin add-on) and the Arctic Fuse skin which is a beautiful piece of work with a fantastic user interface and has great navigation. Hopefully having something similar to that can be achieved in the future

**Comment by Pentaphon** [comment]
Use https://github.com/damontecres/Wholphin if you need an Android TV client other than the official one.

**Comment by keremidarski** [comment]
You are doing great work! Hope the dependency issues are resolved soon and the work on the Apple TV version continues!

**Comment by PokeLu** [comment]
> I believe this should work https://github.com/streamyfin/streamyfin/actions/runs/17856362066/artifacts/4055311355

Do you have any plans for official releases pre-built TV version apk? I tried the deployment recipe given in the README.md but the apk never worked for my TV. It will be great if you could provide the pre-built ones in the releases.

**Comment by Simon-Eklundh** [comment]
> > I believe this should work https://github.com/streamyfin/streamyfin/actions/runs/17856362066/artifacts/4055311355
> 
> Do you have any plans for official releases pre-built TV version apk? I tried the deployment recipe given in the README.md but the apk never worked for my TV. It will be great if you could provide the pre-built ones in the releases.

We do have one in last release (not latest, the one before) 0.30.1

We'll provide proper prebuilds when we get them working tho (and eventually add to the app stores)

**Comment by okseb** [comment]
Any updates on specifically Apple TV support for a newcomer to the thread? (also new to Streamyfin in general and loving it on iOS)

**Comment by Simon-Eklundh** [comment]
I believe we did manage to get appletv to run at least recently 

But we haven't done much tv work

**Comment by iMonZ** [comment]
> I believe we did manage to get appletv to run at least recently
> 
> But we haven't done much tv work

Those are awesome news!
Thank you so much for the work here.

**Comment by enoch85** [comment]
**LG WebOS** support has been requested before. Here's another vote for that. 

What's so hard with developing against their appstore, anything in particular?

**Comment by lancechant** [comment]
> LG WebOS support has been requested before. Here's another vote for that.

Thanks but don't think it will happen **soon**

> What's so hard with developing against their appstore, anything in particular?

Nothing per say is hard to develop it, it's more of OS support, and it's the same as samsung(Tizen) and LG**Web**Os. The big thing there is `Web` both of those only support web apps, which a lot of the packages used, do not support, so we'd have to swap things out to support it, and then package it as a website pretty much, so it's a lot of re-working. So maybe one day, but not right now @enoch85

**Comment by enoch85** [comment]
> > LG WebOS support has been requested before. Here's another vote for that.
> 
> Thanks but don't think it will happen **soon**
> 
> > What's so hard with developing against their appstore, anything in particular?
> 
> Nothing per say is hard to develop it, it's more of OS support, and it's the same as samsung(Tizen) and LG**Web**Os. The big thing there is `Web` both of those only support web apps, which a lot of the packages used, do not support, so we'd have to swap things out to support it, and then package it as a website pretty much, so it's a lot of re-working. So maybe one day, but not right now [@enoch85](https://github.com/enoch85)

Ok! At least it's not a hard "no" and I'm happy with that @lancechant 👍🏼

**Comment by James712346** [comment]
> Any updates on specifically Apple TV support for a newcomer to the thread? (also new to Streamyfin in general and loving it on iOS)

You can install it through a MacBook, by building it, and uploading it to your Apple TV every time you want to use it. It's honestly worth it, as it's more stable and actually usable than the swiftfin app.
Please release it soon, even in the state it's in atm, as it's sooo good.

**Comment by netcelli-tux** [comment]
> > Any updates on specifically Apple TV support for a newcomer to the thread? (also new to Streamyfin in general and loving it on iOS)
> 
> You can install it through a MacBook, by building it, and uploading it to your Apple TV every time you want to use it. It's honestly worth it, as it's more stable and actually usable than the swiftfin app. Please release it soon, even in the state it's in atm, as it's sooo good.

I tried to build from develop branch but when I run `npm run prebuild:tv` it fails with:
```
⚠️ Something went wrong running pod install in the ios directory. Command pod install failed. └─ Cause: The platform of the target Streamyfin (tvOS 15.6) is not compatible with React (0.81.5), which does not support tvOS.
```

**Comment by Simon-Eklundh** [comment]
I think tvos might still be in a dev branch

Not sure tho

**Comment by Rooky030** [comment]
> > Any updates on specifically Apple TV support for a newcomer to the thread? (also new to Streamyfin in general and loving it on iOS)
> 
> You can install it through a MacBook, by building it, and uploading it to your Apple TV every time you want to use it. It's honestly worth it, as it's more stable and actually usable than the swiftfin app.
> Please release it soon, even in the state it's in atm, as it's sooo good.


Can you make a short guide for that? Really want to try it!

**Comment by whoopsi-daisy** [comment]
> > > Any updates on specifically Apple TV support for a newcomer to the thread? (also new to Streamyfin in general and loving it on iOS)
> > 
> > 
> > You can install it through a MacBook, by building it, and uploading it to your Apple TV every time you want to use it. It's honestly worth it, as it's more stable and actually usable than the swiftfin app.
> > Please release it soon, even in the state it's in atm, as it's sooo good.
> 
> Can you make a short guide for that? Really want to try it!

Would advise you to drop by the [Streamyfin Discord](https://discord.streamyfin.app)

**Comment by Gauvino** [comment]
This is done now !

Edit: Now available for tvOS and Android TV

**Comment by tom-james-watson** [comment]
Maybe I'm being dumb or something hasn't updated yet, but the app doesn't appear when searching the playstore on a Chromecast with Google TV (4K). If I try from the website it says it's not compatible.

<img width="563" height="646" alt="Image" src="https://github.com/user-attachments/assets/5772cca4-bf43-4ac5-9f25-16d33c46280a" />

To be clear this is a Google (android) TV stick, not an old-school chromecast.

Is there a trick to installing that I'm missing?

**Comment by Simon-Eklundh** [comment]
> Maybe I'm being dumb or something hasn't updated yet, but the app doesn't appear when searching the playstore on a Chromecast with Google TV (4K). If I try from the website it says it's not compatible.
> 
> <img width="563" height="646" alt="Image" src="https://github.com/user-attachments/assets/5772cca4-bf43-4ac5-9f25-16d33c46280a" />
> 
> To be clear this is a Google (android) TV stick, not an old-school chromecast.
> 
> Is there a trick to installing that I'm missing?

For android TV you still need an apk for now, we're working on the interface tho

**Comment by YESIHACK** [comment]
Does anyone have an update on the Android TV support?

**Comment by Simon-Eklundh** [comment]
@YESIHACK I'd say it's being worked on. One of our Devs recently added exoplayer a an alternate player since TV's are weak, and the UI has been worked on.
